channeldb/payment_control_test.go

```diff
@@ -807,19 +818,24 @@ func TestPaymentControlMultiShard(t *testing.T) {
 			// Record the reason we failed the payment, such that
 			// we can assert this later in the test.
 			firstFailReason = &failReason
-		}
 
-		// The payment should still be considered in-flight, since there
-		// is still an active HTLC.
-		assertPaymentStatus(t, pControl, info.PaymentIdentifier, StatusInFlight)
+			// The payment is now considered pending fail, since
+			// there is still an active HTLC.
+			assertPaymentStatus(
+				t, pControl, info.PaymentIdentifier,
+				StatusInFlight,
+			)
+		}
 
 		// Try to register yet another attempt. This should fail now
 		// that the payment has reached a terminal condition.
 		b = *attempt
 		b.AttemptID = 3
 		_, err = pControl.RegisterAttempt(info.PaymentIdentifier, &b)
-		if err != ErrPaymentTerminal {
-			t.Fatalf("expected ErrPaymentTerminal, got: %v", err)
+		if test.settleFirst {
+			require.Equal(t, err, ErrPaymentAlreadySucceeded)
```

Comment:
nit: same here about order of arguments.

Reply:
fixed.